File: Backend_API/MasanCoffee.API/MasanCoffee.API/Models/NguyenLieu.cs
using System.ComponentModel.DataAnnotations;
namespace MasanCoffee.API.Models
{
    public class NguyenLieu
    {
        [Key]
        public int MaHang { get; set; }
        public required string TenHang { get; set; }
        public int SoLuongTon { get; set; }
        public int MucToiThieu { get; set; }
        public required string DonViTinh { get; set; } 

    }
}

File: Frontend_Angular/src/app/nguyen-lieu.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface NguyenLieu {
  maHang: number;
  tenHang: string;
  soLuongTon: number;
  mucToiThieu: number;
  donViTinh: string;
}

@Injectable({
  providedIn: 'root'
})
export class NguyenLieuService {
  private apiUrl = 'http://localhost:5249/api/NguyenLieu';

  constructor(private http: HttpClient) { }

  getDanhSach(): Observable<NguyenLieu[]> {
    return this.http.get<NguyenLieu[]>(this.apiUrl);
  }
}